ethereum/core/src/main/java/tech/pegasys/teku/core/ForkChoiceUtil.java

```diff
@@ -370,26 +370,30 @@ public static AttestationProcessingResult on_attestation(
    */
   private static AttestationProcessingResult indexAndValidateAttestation(
       MutableStore store, ValidateableAttestation attestation, Checkpoint target) {
-    BeaconState target_state;
+    BeaconState targetState;
     try {
-      target_state = store.getCheckpointState(target);
+      Optional<BeaconState> maybeTargetState = store.getCheckpointState(target);
+      if (maybeTargetState.isEmpty()) {
+        return AttestationProcessingResult.UNKNOWN_BLOCK;
+      }
+      targetState = maybeTargetState.get();
     } catch (final InvalidCheckpointException e) {
       LOG.debug("on_attestation: Attestation target checkpoint is invalid", e);
       return AttestationProcessingResult.invalid("Invalid target checkpoint: " + e.getMessage());
     }
 
     // Get state at the `target` to validate attestation and calculate the committees
```

Comment:
nit: We might as well remove this stale "get state at the target" comment, or move it up to where it's appropriate.

Reply:
Done.